src/wallet/rpcwallet.cpp

```diff
@@ -598,6 +652,102 @@ UniValue listaddresses(const UniValue& params, bool fHelp)
         entry.pushKV("source", "legacy_hdseed");
 
         bool hasData = add_sapling(saplingAddresses, PaymentAddressSource::LegacyHDSeed, entry);
+        if (hasData) {
+            ret.push_back(entry);
+        }
+    }
+
+    // mnemonic seed source
+    {
+        UniValue entry(UniValue::VOBJ);
+        entry.pushKV("source", "mnemonic_seed");
+
+        bool hasData = false;
+        // transparent
+        // For each address in pwalletMain->mapKeyMetadata, include any address
+        // for which its HD keypath metadata indicates it is associated with the legacy
+        // account ID. All such addresses are derived from the mnemonic seed.
+        UniValue mnemonic_taddrs(UniValue::VARR);
+        for (const auto& [keyId, keyMeta] : pwalletMain->mapKeyMetadata) {
+            auto account = libzcash::ParseHDKeypathAccount(32, Params().BIP44CoinType(), keyMeta.hdKeypath);
```

Comment:
```suggestion
            auto account = libzcash::ParseHDKeypathAccount(44, Params().BIP44CoinType(), keyMeta.hdKeypath);
```
Transparent address keypaths continue to use 44 as the purpose.